src/components/CodeAssignmentModal/CodeAssignmentModal.test.jsx

```diff
@@ -0,0 +1,178 @@
+import CodeAssignmentModal, { BaseCodeAssignmentModal } from '.';
+import { ASSIGNMENT_MODAL_FIELDS } from './constants';
+import { displayCode, displaySelectedCodes } from '../CodeModal/codeModalHelpers';
+
+import {
+  EMAIL_TEMPLATE_SOURCE_NEW_EMAIL,
+} from '../../data/constants/emailTemplate';
+import { EMAIL_FORM_NAME } from '../EmailTemplateForm';
+
+const sampleCodeData = {
+  code: 'test-code-1',
+  assigned_to: 'test@bestrun.com',
+  redemptions: {
+    total: 100,
+    used: 10,
+  },
+  error: null,
+};
+const sampleTableData = {
+  loading: false,
+  error: null,
+  data: {
+    count: 3,
+    num_pages: 2,
+    current_page: 1,
+    results: [
+      sampleCodeData,
+      {
+        ...sampleCodeData,
+        code: 'test-code-2',
+        redemptions: {
+          total: 100,
+          used: 100,
+        },
+      },
+      {
+        ...sampleCodeData,
+        code: 'test-code-3',
+        assigned_to: null,
+      },
+    ],
+  },
+};
+
+const initialProps = {
+  handleSubmit: () => {},
+  submitting: false,
+  submitSucceeded: false,
+  submitFailed: false,
+  error: [],
+  couponId: 12,
+  title: 'Remind me',
+  onClose: () => {},
+  onSuccess: () => {},
+  sendCodeAssignment: () => {},
+  couponDetailsTable: {
+    data: {
+      count: 1,
+    },
+  },
+  initialValues: {},
+  isBulkRemind: false,
+  selectedToggle: 'toggle',
+  data: {
+    selectedCodes: [{ code: 'bulk' }],
+    unassignedCodes: 4,
+    code: { code: 'ind' },
+    remainingUses: 3,
+    hasAllCodesSelected: false,
+  },
+  enableLearnerPortal: false,
+  enterpriseSlug: 'bearsRus',
+  setEmailAddress: () => {},
+  enterpriseUuid: 'foo',
+  createPendingEnterpriseUsers: () => {},
+};
+
+const mockStore = configureMockStore([thunk]);
+
+const initialState = {
+  table: {
+    'coupon-details': sampleTableData,
+  },
+  emailTemplate: {
+    loading: false,
+    error: null,
+    emailTemplateSource: EMAIL_TEMPLATE_SOURCE_NEW_EMAIL,
+    default: {
+      assign: {
+        'email-template-subject': remindEmailTemplate.subject,
+        'email-template-greeting': remindEmailTemplate.greeting,
+        'email-template-body': remindEmailTemplate.body,
+        'email-template-closing': remindEmailTemplate.closing,
+      },
+    },
+    assign: {
+      'email-template-subject': remindEmailTemplate.subject,
+      'email-template-greeting': remindEmailTemplate.greeting,
+      'email-template-body': remindEmailTemplate.body,
+      'email-template-closing': remindEmailTemplate.closing,
+    },
+  },
+};
+
+/* eslint-disable react/prop-types */
+const CodeAssignmentModalWrapper = (props) => (
+  <MemoryRouter>
+    <Provider store={mockStore(initialState)}>
+      <CodeAssignmentModal
+        {...initialProps}
+        {...props}
+      />
+    </Provider>
+  </MemoryRouter>
+);
+/* eslint-enable react/prop-types */
+
+describe('CodeAssignmentModal component', () => {
+  // beforeEach(() => jest.resetModules());
```

Comment:
still not needed?

Reply:
Possibly needed if I can get the test working.